### ISSUE TITLE
Coding error in GNTPP/models/prob_decoders/vae_modules /vae.py

### ISSUE THREAD

**Issue by AnthonyChouGit** [BIRD-TAO/GNTPP#4]
There is an error in file  [vae.py](https://github.com/BIRD-TAO/GNTPP/blob/main/models/prob_decoders/vae_modules/vae.py). At the version on May 6, 2023, the bug is in line 131, where KL-divergence term is computed. It reads as follows:
`kl_loss = (- 0.5 * torch.sum(1 + log_var - mean.pow(2) - log_var.exp())) * seq_onehots`
Note that the "dim" parameter is passed to torch.sum(), which means this operation is summing over the whole tensor "1 + log_var - mean.pow(2) - log_var.exp()", yielding a scalar! No exception is raised, because multiplying this scalar to the onehot mask is still valid and it preserves the tensor shape. But this is not what we wanna do to compute the loss. This line should be refactored as:
`kl_loss = (- 0.5 * torch.sum(1 + log_var - mean.pow(2) - log_var.exp(), dim=-1)) * seq_onehots`

Please check it out.

**Comment by EDAPINENUT** [comment]
The bug is fixed, and the new experimental results will be updated in the Arxiv version. Thanks for your following.

**Comment by EDAPINENUT** [comment]
Fixed
